sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/BlobDecryptionPolicy.java

```diff
@@ -151,70 +247,23 @@ Flux<ByteBuffer> decryptBlob(String encryptedDataString, Flux<ByteBuffer> encryp
         } else {
             dataToTrim = getKeyEncryptionKey(encryptionData)
                 .flatMapMany(contentEncryptionKey -> {
-                    /*
-                     * Calculate the IV.
-                     *
-                     * If we are starting at the beginning, we can grab the IV from the encryptionData. Otherwise,
-                     * Reactor makes it difficult to grab the first 16 bytes of data to pass as an IV to the cipher.
-                     * As a workaround, we initialize the cipher with a garbage IV (empty byte array) and attempt to
-                     * decrypt the first 16 bytes (the actual IV for the relevant data). We throw away this "decrypted"
-                     * data. Now, though, because each block of 16 is used as the IV for the next, the original 16 bytes
-                     * of downloaded data are in position to be used as the IV for the data actually requested and we
-                     * are in the desired state.
-                     */
-                    byte[] iv;
-
-                    /*
-                     * Adjusting the range by <= 16 means we only adjusted to align on an encryption block boundary
-                     * (padding will add 1-16 bytes as it will prefer to pad 16 bytes instead of 0) and therefore the
-                     * key is in the metadata.
-                     */
-                    if (encryptedBlobRange.getOffsetAdjustment() <= ENCRYPTION_BLOCK_SIZE) {
-                        iv = encryptionData.getContentEncryptionIV();
-                    } else {
-                        iv = new byte[ENCRYPTION_BLOCK_SIZE];
-                    }
-
-                    Cipher cipher;
-                    try {
-                        cipher = getCipher(contentEncryptionKey, encryptionData, iv, padding);
-                    } catch (InvalidKeyException e) {
-                        throw LOGGER.logExceptionAsError(Exceptions.propagate(e));
+                    switch (encryptionData.getEncryptionAgent().getProtocol()) {
```

Comment:
This value is guaranteed to be non-null, right? If not, I'd suggest using if/else if/else blocks as this will throw a very confusing NullPointerException if it can be null (which has been fun to debug in the past)

Reply:
Unless someone has tampered with the metadata, it'll never be null. We always set the protocol version.